components/dashboard/src/admin/WorkspacesSearch.tsx

```diff
@@ -66,6 +67,15 @@ export function WorkspaceSearch(props: Props) {
     }
 
     const search = async () => {
+        if (!props.user) { // In the workspace search page
+            // This only allows searching for exact-matches of workspace or instance IDs
+            if ((queryTerm.length === 0) ||
+                (!matchesNewWorkspaceIdExactly(queryTerm) &&
```

Comment:
IMO we should try to avoid calling the `matches` functions multiple times.

One way do to that is to move it too line 79 :point_down:

Reply:
Yes, good point!

Reply:
I was thinking we could return early when there is an empty search, and then check further down if they're partial matches, like so:

```
    const search = async () => {
        // Disables empty search on the workspace search page
        if (!props.user && queryTerm.length === 0) {
            return;
        }

        setSearching(true);
        try {
            ...
            if (!query.ownerId && !matchesNewWorkspaceIdExactly(queryTerm) && !matchesInstanceIdOrLegacyWorkspaceIdExactly(queryTerm)) {
                return;
            }
```

What do you think?